### PR TITLE
airport_challenge

### PR DESCRIPTION
**Pull request by emilyworrall**

### CODE REVIEW

File: spec/plane_spec.rb [hunk 1]
```diff
@@ -1,30 +1,36 @@
...
-
+  subject {Plane.new}
+
+  it 'shows flying status when created' do
+  	expect(subject.flying?).to eql(true)
+  end
+
+  describe 'landing' do
+    it 'can land' do
+    	subject.taken_off
```
Comment (natstar93) [inline spec/plane_spec.rb]: Change to take_off

File: spec/weather_spec.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    plane.landed
+    expect {airport.land(plane)}.to raise_error 'Weather is stormy'
+  end
+
+  it 'will not let a plane take off if stormy' do
+    plane.landed
+    allow(airport).to receive(:stormy?) {true}
+    expect {airport.take_off(plane)}.to raise_error 'Weather is stormy'
+  end
+end
```
Comment (natstar93) [inline spec/weather_spec.rb]: Can move these 2 tests to airport spec

File: lib/plane.rb [hunk 3]
```diff
@@ -1,2 +1,19 @@
...
+  end
+
+  def landed
+  	@flying = false
+  end
+
+  def taken_off
+  	@flying = true
+  end
+
```
Comment (natstar93) [inline lib/plane.rb]: Change method names to 'land' and 'take_off'

Indentation

**Comment by natstar93** [comment]
Good work! :star:

File: lib/airport.rb [hunk 4]
```diff
@@ -1,2 +1,41 @@
...
+  end
+
+  def full?
+    plane_total == @capacity
+  end
+
+  def empty?
+    plane_total == 0
+  end
+
```
Comment (spike01) [inline lib/airport.rb]: Usually you don't need this extra line break at the bottom of your class

File: spec/airport_spec.rb [hunk 5]
```diff
@@ -1,48 +1,54 @@
...
 
-# A plane currently in the airport can be requested to take off.
-#
-# No more planes can be added to the airport, if it's full.
-# It is up to you how many planes can land in the airport
-# and how that is implemented.
-#
-# If the airport is full then no planes can land
+  let(:subject) {Airport.new}
+  let(:plane) {Plane.new}
```
Comment (spike01) [inline spec/airport_spec.rb]: Try to use doubles!

File: spec/airport_spec.rb [hunk 6]
```diff
@@ -1,48 +1,54 @@
...
 
-describe Airport do
+  before do
+    allow(subject).to receive(:stormy?) {false}
```
Comment (spike01) [inline spec/airport_spec.rb]: Prefer using `and_return` over the block syntax

File: spec/airport_spec.rb [hunk 7]
```diff
@@ -1,48 +1,54 @@
...
+      expect {subject.land(plane)}.to change {subject.plane_total}.by (1)
+    end
   end
 
   describe 'traffic control' do
-    context 'when airport is full' do
-      xit 'does not allow a plane to land'
+    it 'does not allow a plane to land when airport at capacity' do
+      5.times {subject.land(plane)}
+      expect {subject.land(plane)}.to raise_error 'Airport at capacity'
```
Comment (spike01) [inline spec/airport_spec.rb]: Try to be consistent with parentheses around your arguments - in the two previous tests you've used them, but then suddenly drop them here

File: spec/plane_spec.rb [hunk 8]
```diff
@@ -1,30 +1,36 @@
...
-
+  subject {Plane.new}
+
+  it 'shows flying status when created' do
+    expect(subject.flying?).to eql(true)
+  end
+
+  describe 'landing' do
+    it 'can land' do
+      subject.take_off
```
Comment (spike01) [inline spec/plane_spec.rb]: You don't really need this line - you know (from the test above) that the plane is already flying!

**Comment by spike01** [comment]
Is your Weather module tested? Otherwise good submission - just a few small things. :checkered_flag:

**Pull request closed without merging**